WriteVHDLSyntax.py

```diff
@@ -89,136 +95,245 @@ def writeTBMemoryReadInstance(memModule):
+        mem_str += "    "+mtypeB+" : entity work.tf_mem\n"        
+    mem_str += "      generic map (\n"+parameterlist.rstrip(",\n")+"\n      )\n"
+    mem_str += "      port map (\n"+portlist.rstrip(",\n")+"\n      );\n\n"
+    mem_str += "  end generate "+genName+";\n\n\n"
+
     return wirelist,mem_str
 
-def writeControlSignals_interface(initial_proc, final_proc):
+def writeControlSignals_interface(initial_proc, final_proc, notfinal_procs):
     """
     # Top-level interface: control signals
     """
     string_ctrl_signals = ""
     string_ctrl_signals += "    clk        : in std_logic;\n"
     string_ctrl_signals += "    reset      : in std_logic;\n"
     string_ctrl_signals += "    "+initial_proc+"_start  : in std_logic;\n"
-    string_ctrl_signals += "    bx_in_"+initial_proc+" : in std_logic_vector(2 downto 0);\n"
-    string_ctrl_signals += "    bx_out_"+final_proc+" : out std_logic_vector(2 downto 0);\n"
-    string_ctrl_signals += "    bx_out_"+final_proc+"_vld : out std_logic;\n"
+    string_ctrl_signals += "    "+initial_proc+"_bx_in : in std_logic_vector(2 downto 0);\n"
+    string_ctrl_signals += "    "+final_proc+"_bx_out : out std_logic_vector(2 downto 0);\n"
+    string_ctrl_signals += "    "+final_proc+"_bx_out_vld : out std_logic;\n"
     string_ctrl_signals += "    "+final_proc+"_done   : out std_logic;\n"
+    # Extra output ports if debug info must be sent to test-bench.
+    for mid_proc in notfinal_procs:
+        string_ctrl_signals += "    "+mid_proc+"_bx_out : out std_logic_vector(2 downto 0);\n"
+        string_ctrl_signals += "    "+mid_proc+"_bx_out_vld : out std_logic;\n"
+        string_ctrl_signals += "    "+mid_proc+"_done   : out std_logic;\n"
+
     return string_ctrl_signals
 
-def writeMemoryLHSPorts_interface(memModule):
+def writeMemoryLHSPorts_interface(mtypeB, extraports=False):
     """
     # Top-level interface: input memories' ports.
     """
+
+    if (extraports):
+        direction = "out" # carry debug info to test-bench
+    else:
+        direction = "in"
+
     string_input_mems = ""
-    string_input_mems += "    "+memModule.inst+"_dataarray_data_V_wea       : in std_logic;\n"
-    string_input_mems += "    "+memModule.inst+"_dataarray_data_V_writeaddr : in std_logic_vector("
-    string_input_mems += str(6+memModule.bxbitwidth)+" downto 0);\n"
-    string_input_mems += "    "+memModule.inst+"_dataarray_data_V_din       : in std_logic_vector("
-    string_input_mems += str(memModule.bitwidth-1)+" downto 0);\n"
+    string_input_mems += "    "+mtypeB+"_mem_A_wea        : "+direction+" t_arr_"+mtypeB+"_1b;\n"
+    string_input_mems += "    "+mtypeB+"_mem_AV_writeaddr : "+direction+" t_arr_"+mtypeB+"_ADDR;\n"
+    string_input_mems += "    "+mtypeB+"_mem_AV_din       : "+direction+" t_arr_"+mtypeB+"_DATA;\n"
+
     return string_input_mems
 
-def writeMemoryRHSPorts_interface(memModule):
+def writeMemoryRHSPorts_interface(mtypeB, memList, memInfo):
```

Comment:
memList doesn't appear to be used in this function

Reply:
Agreed. This unused function argument now removed.